lib/cylc/param_expand.py

```diff
@@ -210,25 +210,30 @@ def _expand_name(self, str_tmpl, param_list, results, spec_vals=None):
     def replace_params(self, name_in, param_values, origin):
         """Replace parameters in name_in with values in param_values.
 
-        Note this is "expansion" for specific values, not all values.
+        This is "expansion" for specific values, not for all values.
         """
         name, p_tmpl = REC_P_NAME.match(name_in).groups()[:2]
         if not p_tmpl:
             # name_in is not parameterized.
             return name_in
-        # List of parameter names used in this name: ['m', 'n']
-        used_param_names = [i.strip() for i in p_tmpl[1:-1].split(',')]
-        for p_name in used_param_names:
-            msg = None
-            if '=' in p_name:
-                msg = 'values'
-            elif '-' in p_name or '+' in p_name:
-                msg = 'offsets'
-            if msg is not None:
-                raise ParamExpandError("ERROR, parameter %s not supported"
-                                       " here: %s" % (msg, origin))
+        used_params = [i.strip() for i in p_tmpl[1:-1].split(',')]
```

Comment:
I wonder if it is worth adjusting what `REC_P_NAME` captures so we don't have to do `p_tmpl[1:-1]`. (`REC_P_NAME` is used in another location with a similar logic.)

Reply:
I think the code will be easier to understand if we have something like `captured_params` and `used_params`.